dev-docs/RFCs/v8.1/gpu-accelerated-geo-fencing.md

```diff
@@ -0,0 +1,153 @@
+
+Entire process of filtering can be divided into two steps, 1. Generation of filter texture and 2. performing actual filtering.
+
+* API : Following new components will be added
+
+#### GeoFilterEffect
+
+This effect generates the required filter texture. Internally it creates a new render pass to render data in world space to a offline texture.
+
+During pre render pass:
+- All layers are processed, and if a layer has 'geoFilterMask' prop set, a bounding box of its data is calculated (optionally can be provided by the layer).
+- All above bounding boxes are merged at each step and used to build the texture mask.
+- This effect class is responsible for providing following uniforms that will be consumed by following shader module 'TextureFilterModule'
+
+1. `textureTransform` (Array, [xTranslation, yTranslation, xScale, yScale]) : used to transform each data point into required clipping space.
+2. `maskTexture` (Texture2D) : texture that is sampled to determine if a given point is inside or outside.
+
+#### TextureFilterModule
+
+A new shader module is added to perform sampling from the texture and filter the points, defines following shader functions:
+
+- `geoFilter_setValue` : Takes, object's world positions, transforms it to mask texture space, performs sampling and sets `geoFilter_value` to either 0 (if outside the defined region) or to and id ( >= 1).
+
+Also this module defined following shader injects/hooks, that can be used by layer shader to perform geo filtering.
+
+- `vs:#main-start` : To perform sampling and set `getoFilter_value`.
+- `vs:DECKGL_FILTER_SIZE` : To set object size to 0 if the object is filteredout.
+- `fs:DECKGL_FILTER_COLOR` : To discard the fragments if filter value is 0. This can also be used update the color of the fragment if it is inside the region.
+
+
+Sample code to perform filtering on a `ScatterplotLayer` to a `SolidPolygonLayer` that is rendering a complex polygon.
+
+```js
+import DeckGL from '@deck.gl/react';
+import {GeoFilterEffect} from '@deck.gl/core';
+
+const GEO_FILTER_ID = 'geo-filter-id-1'
+
+const filterEffect = new GeoFilterEffect({id: GEO_FILTER_ID, precession: 0.8});
+
+// ...
+
+const App = (data) => (
+  <DeckGL
+    layers={[
+      // Layer that generates the mask
+      new SolidPolygonLayer({
+        // Complex polygon with one hole
+        data: [
+              [[0, 0], [0, 2], [2, 2], [2, 0], [0, 0]],
+              [[0, 0], [0, 1], [1, 1], [1, 0], [0, 0]]
+            ],
+        visible: false
+        geoFilterMask: GEO_FILTER_ID
+        // ...
+      })
+      // Layer that gets filtered out
+      new ScatterplotLayer({
+        data,
+        geoFilter: GEO_FILTER_ID
```

Comment:
Do you allow a layer to be filtered by multiple masks?

Reply:
Yeah that was the original idea. I might rename this this to `MASK_ID`. One ore more layers can contribute to a given `MASK_ID` and a given layer can be masked by one ore more `MASK_ID`, but for initial iteration, I will limit this to one MASK. Will make this prop (call it something like `filterToMask`) a boolean and in later revisions this can be an array of mask Ids.